### PR TITLE
BGA-149 Sonar issues

### PR DESCRIPTION
**Pull request by gonzaalvarez70**
Solved the following issues

src/coin/cgld/stakingBuilder.ts
Function has a complexity of 11 which is greater than 10 authorized

src/coin/cgld/types.ts
Function has a complexity of 13 which is greater than 10 authorized.

src/coin/eth/transactionBuilder.ts
Function has a complexity of 15 which is greater than 10 authorized.

src/coin/eth/utils.ts
Function has a complexity of 11 which is greater than 10 authorized.

### CODE REVIEW

File: src/coin/eth/transactionBuilder.ts [hunk 1]
```diff
@@ -260,6 +243,30 @@ export class TransactionBuilder extends BaseTransactionBuilder {
     }
   }
 
+  private validateWalletInitializationFields(): void {
```
Comment (argjv) [inline src/coin/eth/transactionBuilder.ts:246]: Add a short JS doc for this and other methods please:
`/** Check mandatory fields for wallet initialization transactions are valid or throw.  */`

File: src/coin/eth/transactionBuilder.ts [hunk 2]
```diff
@@ -260,6 +243,30 @@ export class TransactionBuilder extends BaseTransactionBuilder {
     }
   }
 
+  private validateWalletInitializationFields(): void {
+    if (this._walletOwnerAddresses === undefined) {
+      throw new BuildTransactionError('Invalid transaction: missing wallet owners');
+    }
+
+    if (this._walletOwnerAddresses.length !== 3) {
```
Comment (twtaylorbitgo) [inline src/coin/eth/transactionBuilder.ts:251]: is it possible this is null?

File: src/coin/eth/transactionBuilder.ts [hunk 3]
```diff
@@ -260,6 +243,30 @@ export class TransactionBuilder extends BaseTransactionBuilder {
...
   }
 
+  private validateWalletInitializationFields(): void {
+    if (this._walletOwnerAddresses === undefined) {
+      throw new BuildTransactionError('Invalid transaction: missing wallet owners');
+    }
+
+    if (this._walletOwnerAddresses.length !== 3) {
+      throw new BuildTransactionError(
+        `Invalid transaction: wrong number of owners -- required: 3, ` + `found: ${this._walletOwnerAddresses.length}`,
```
[the remark this answers is not in the document]
Comment (gonzaalvarez70) [reply src/coin/eth/transactionBuilder.ts:253]: It's unused, removing it on the next commit.

File: src/coin/eth/transactionBuilder.ts (hunk 1 above)
Comment (gonzaalvarez70) [reply src/coin/eth/transactionBuilder.ts:246]: Sure, that sounds good.

File: src/coin/eth/transactionBuilder.ts (hunk 2 above)
Comment (gonzaalvarez70) [reply src/coin/eth/transactionBuilder.ts:251]: It shouldn't be possible, it just could be undefined and that is validated on the previous lines.

**Pull request merged**